Sources/libhostmgr/BuildkiteScriptBuilder.swift

```diff
@@ -0,0 +1,165 @@
+import Foundation
+import TSCBasic
+
+public struct BuildkiteScriptBuilder {
+
+    /// A list of files to run `source` against prior to executing the rest of the script.
+    var dependencies = [String]()
+
+    /// A list of key/value pairs representing environment variables that should be defined at the start of the script.
+    var environmentVariables: [String: Value] = [:]
+
+    /// A list of commands to run in the script.
+    var commands = [Command]()
+
+    public init() {
+        self.environmentVariables["BUILDKITE"] = Value(wrapping: "true")
```

Comment:
Any reason why this one is set in the `BuildkiteScriptBuilder`'s `init` here, while all the others (like `CI=true` or `BUILDKITE_AGENT_NAME=builder`) are set at usage site (in `GenerateBuildkiteJobScript.swift`)?

Makes us having multiple places / source of truth for the list of env vars the script will contain, and thus imho harder to maintain compared to having all the ones we want to declare all listed in a single place in `GenerateBuildkiteJobScript.swift`.